### PR TITLE
Add ts definition file

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
#32 

I'm unsure what to define as this is a fastify plugin. The plugin only has 1 new option property which is `bodyLimit` and it is of type Number. Where would I add this in the definition? In the namespace interface? 

I based this off of the type file in [`fastify-cookie`](https://github.com/fastify/fastify-cookie/blob/master/plugin.d.ts)

### CODE REVIEW

**Comment by mcollina** [comment]
You should also add unit test, see https://github.com/fastify/fastify-cookie/blob/master/test/types.test.ts.


I don't think options could be defined anywhere in plugins. Specifically, options are plugin specific but the type for that object is https://github.com/fastify/fastify/blob/master/fastify.d.ts#L228-L231.
I don't know if we could change `register` types so that a plugin could specify its own type. Or if you could add an overload of `register` in this module so that if fastify-cookie is passed in, a specific options object is used. If this is a viable strategy, then we should also add it to the fastify docs.

cc @evanshortiss @delvedor

**Comment by Ethan-Arrowood** [comment]
What I don't understand (and this is probably due to my lack of experience with TS), but why are the types defined in `fastify-plugin.d.ts` not inherited in the plugin js files? For example, in this repo, when i hover over `fp` I get typescript docs that include a bunch of `any` where those types are definitely defined in `fastify-plugin`. 
screen shot 2019-01-11 at 6 53 27 pm

**Comment by mcollina** [comment]
I think this one of the most complete: https://github.com/fastify/fastify-swagger/blob/master/index.d.ts. Options are set as well.

**Comment by Ethan-Arrowood** [comment]
I reverted the version bump changes I made but something is still wrong with package.json files 😕

**Comment by Ethan-Arrowood** [comment]
This was the error:
screen shot 2019-01-13 at 11 06 14 pm

Removing lock file fixes it.

File: formbody.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import * as fastify from 'fastify'
+
+declare namespace formBodyPlugin {
+  interface FormBodyPluginOptions {
+    bodyLimit?: number
+  }
+}
+
+declare let formBodyPlugin: fastify.Plugin<
+  http.Server,
```
Comment (mysticatea) [inline formbody.d.ts]: Hi.

I have encountered a type error: "Type 'import("http").Server' is not assignable to type 'import("https").Server'."
Is there a way to use this plugin with https?

Comment (Ethan-Arrowood) [reply formbody.d.ts]: Interesting, I thought the https.Server extended from http.Server.

Try adding the https parameters as unions to this definition and see if that works:

```ts
declare let formBodyPlugin: fastify.Plugin<
  http.Server | https.Server,
  ...
>
```

IF that works, would you like to open a PR with the updates?

**Pull request merged**